x-pack/legacy/plugins/infra/public/pages/logs/analysis/setup/analysis_setup_indices_form.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo, useCallback } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { EuiCode, EuiDescribedFormGroup, EuiFormRow, EuiCheckboxGroup } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+const indicesSelectionLabel = i18n.translate('xpack.infra.analysisSetup.indicesSelectionLabel', {
+  defaultMessage: 'Indices',
+});
+
+type IndicesSelection = Record<string, boolean>;
+
+export const AnalysisSetupIndicesForm: React.FunctionComponent<{
+  indices: IndicesSelection;
+  onChangeSelectedIndices: (selectedIndices: IndicesSelection) => void;
+  validationErrors?: Array<'TOO_FEW_SELECTED_INDICES'>;
+}> = ({ indices, onChangeSelectedIndices, validationErrors = [] }) => {
+  const choices = useMemo(
+    () =>
+      Object.keys(indices).map(indexName => ({
+        id: indexName,
+        label: <EuiCode>{indexName}</EuiCode>,
+      })),
+    [indices]
+  );
+
+  const handleCheckboxGroupChange = useCallback(
+    indexName => {
+      onChangeSelectedIndices({
+        ...indices,
+        [indexName]: !indices[indexName],
+      });
+    },
+    [indices, onChangeSelectedIndices]
+  );
+
+  return (
+    <EuiDescribedFormGroup
+      idAria="indices"
+      title={
+        <FormattedMessage
+          id="xpack.infra.analysisSetup.indicesSelectionTitle"
+          defaultMessage="Choose indices"
+        />
+      }
+      description={
+        <FormattedMessage
+          id="xpack.infra.analysisSetup.indicesSelectionDescription"
+          defaultMessage="By default, Machine Learning analyzes log messages in all log indices configured for the source. You can choose to only analyse a subset of the index names. Every selected index name must match at least one index with log entries."
```

Comment:
"...only analyse" => "...only analyZe", to keep it consistent

Reply:
Good catch... as a non-native speaker this always trips me up